lib/util.js

```diff
@@ -478,7 +478,22 @@ function formatValue(ctx, value, recurseTimes) {
   if (ctx.showHidden) {
     keys = Object.getOwnPropertyNames(value);
   } else {
-    keys = Object.keys(value);
+    // This might throw if `value` is a Module Namespace Object from an
+    // unevaluated module, but we don't want to perform the actual type
+    // check because it's expensive.
+    // TODO(devsnek): track https://github.com/tc39/ecma262/issues/1209
+    // and modify this logic as needed.
+    try {
+      keys = Object.keys(value);
+    } catch (err) {
+      if (err instanceof ReferenceError &&
+          types.isModuleNamespaceObject(value)) {
+        keys = Object.getOwnPropertyNames(value);
+      } else {
```

Comment:
\cc @bmeurer A pattern in much of Node's code is to extract try-catches into their own functions. In this case it would be named `tryKeys()`. Is this practice still beneficial in the V8 of today?

Reply:
iirc it hasn't been needed since like v6

Reply:
That is not necessary anymore. It used to prevent a function from optimizing and this is not the case anymore with turbofan.